corehq/apps/hqwebapp/views.py

```diff
@@ -1262,9 +1265,27 @@ def page_url(self):
 @require_superuser
 def create_alert(request):
     from corehq.apps.hqwebapp.models import MaintenanceAlert
+    import pytz
     alert_text = request.POST.get('alert_text')
     domains = request.POST.get('domains').split() or None
-    MaintenanceAlert(active=False, text=alert_text, domains=domains).save()
+
+    start_time = request.POST.get('start_time') or None
+    end_time = request.POST.get('end_time') or None
+    timezone = request.POST.get('timezone') or 'UTC'
+
+    if start_time and end_time:
+        start_time = UserTime(
+            datetime.fromisoformat(start_time), tzinfo=pytz.timezone(timezone)
+        ).server_time().ui_string()
+        end_time = UserTime(
+            datetime.fromisoformat(end_time), tzinfo=pytz.timezone(timezone)
+        ).server_time().ui_string()
+    else:
+        start_time = None
+        end_time = None
```

Comment:
I wonder if users should get some kind of warning/error here since they are expecting to create an alert that starts and stops some time in the future. I'm imagining someone not noticing that they neglected to input either a start or end time and thinking the alert will activate at the expected time since it was created successfully.

Reply:
This condition shouldn't be reached in normal operation, right?  If that's the case, I'd say it shouldn't swallow the exception, and should instead just 500.  I imagine the issue would be something like a bad timezone or datetime format, and we'd want to see those details so we can fix it.